Sources/OpenAI/Public/Models/ModerationsResult.swift

```diff
@@ -26,25 +35,93 @@ public struct ModerationsResult: Codable, Equatable {
             public let violence: Bool
             /// Violent content that depicts death, violence, or serious physical injury in extreme graphic detail.
             public let violenceGraphic: Bool
-            
-            enum CodingKeys: String, CodingKey {
+
+            public enum CodingKeys: String, CodingKey {
+                case harassment
+                case harassmentThreatening = "harassment/threatening"
                 case hate
                 case hateThreatening = "hate/threatening"
                 case selfHarm = "self-harm"
+                case selfHarmIntent = "self-harm/intent"
+                case selfHarmInstructions = "self-harm/instructions"
                 case sexual
                 case sexualMinors = "sexual/minors"
                 case violence
                 case violenceGraphic = "violence/graphic"
             }
+
+            public func makeIterator() -> CategoriesIterator {
+                return CategoriesIterator(self)
+            }
+
+            public struct CategoriesIterator: IteratorProtocol {
```

Comment:
Hi, addressing your question:
> got a more concise way to do the Iterator? Reflection is no longer a concise way to flit through these.
  
What do you think of perhaps using Mirror? Maybe something like this:
```swift
public struct CategoriesIterator: IteratorProtocol {
     public typealias Element = (String, Bool)

      private let categories: Categories
      private var properties: [(String, Bool)]

      public init(_ categories: Categories) {
           self.categories = categories
           self.properties = Mirror(reflecting: categories).children.compactMap {
                 guard let name = $0.label, let value = $0.value as? Bool else { return nil }
                 return (name, value)
           }
       }

       public mutating func next() -> Element? {
            guard !properties.isEmpty else { return nil }
            return properties.removeFirst()
       }
}
```

Reply:
@ingvarus-bc , good thought. The trouble with doing it that way is, as far as I can determine, it can only reflect the camelCase names, and not the CodingKeys translations. Hence the long form... Unless you can figure out a way of looking up the CodingKeys from the camelCase names, which I couldn't.

Reply:
OK, got it with reflection. Relying on the order of CodingKeys being same as the order of variables in the structs. Checking this with DEBUG asserts for future verification.